rust/src/nfs/nfs.rs

```diff
@@ -438,6 +455,85 @@ impl NFSState {
         }
     }
 
+    fn add_rpc_udp_ts_frames(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], rpc_len: i64) -> (Option<Frame>, Option<Frame>) {
+        let rpc_udp_ts_pdu = Frame::new_ts(flow, stream_slice, input, rpc_len, NFSFrameType::RPCPdu as u8);
+        let rpc_udp_ts_creds = Frame::new_ts(flow, stream_slice, &input[24..], rpc_len - 32, NFSFrameType::RPCCreds as u8);
+        SCLogDebug!("RPC_UDP To Server Frames {:?} {:?}", rpc_udp_ts_pdu, rpc_udp_ts_creds);
+        (rpc_udp_ts_pdu, rpc_udp_ts_creds)
+    }
+
+    fn add_rpc_tcp_ts_frames(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], rpc_len: i64) -> (Option<Frame>, Option<Frame>) {
+        let rpc_tcp_ts_pdu = Frame::new_ts(flow, stream_slice, input, rpc_len, NFSFrameType::RPCPdu as u8);
+        let rpc_tcp_ts_creds = Frame::new_ts(flow, stream_slice, &input[28..], rpc_len - 36, NFSFrameType::RPCCreds as u8);
+        SCLogDebug!("RPC_TCP To Server Frames {:?} {:?}", rpc_tcp_ts_pdu, rpc_tcp_ts_creds);
+        (rpc_tcp_ts_pdu,rpc_tcp_ts_creds)
+    }
+
+    fn add_nfs_ts_frame(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], nfs_req_len: i64) -> Option<Frame> {
+        let nfs_req_pdu = Frame::new_ts(flow, stream_slice, input, nfs_req_len, NFSFrameType::NFSPdu as u8);
+        SCLogDebug!("NFS Request Frame {:?}", nfs_req_pdu);
+        nfs_req_pdu
+    }
+
+    fn add_nfs4_ts_pdu_frame(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], nfs4_req_len: i64) -> Option<Frame> {
+        let nfs4_ts_pdu = Frame::new_ts(flow, stream_slice, input, nfs4_req_len, NFSFrameType::NFS4Pdu as u8);
+        SCLogDebug!("NFS4 Request Frame: {:?}", nfs4_ts_pdu);
+        nfs4_ts_pdu
+    }
+
+    fn add_nfs4_ts_hdr_ops_frame(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], nfs4_req_len: i64) {
+        if nfs4_req_len > 0 {
+            let _nfs4_ts_hdr = Frame::new_ts(flow, stream_slice, input, 8, NFSFrameType::NFS4Hdr as u8);
+            let _nfs4_ts_ops = Frame::new_ts(flow, stream_slice, &input[8..], nfs4_req_len - 8, NFSFrameType::NFS4Ops as u8);
```

Comment:
Correct that's probably due to NULL records .
Thanks! On it.

Reply:
So, null records which enter a block that checks for len>0 ? That sounds like the case of gaps. How do frames deal with it? Do frames take care of it already?

Reply:
Nope that was a rush assumption. NFS4 NULL records don't hold a header.
This is a matter of having different header sizes, does the Fuzzing Checks run on the S-V tests pcaps?

Reply:
No. They run on a public corpus.. I don't think they have anything to do w s-v

Reply:
I meant the pcaps as input.
So I don't have an option to look at the buffer the check failed against?
Is there a way to run the fuzzer locally?

Reply:
It doesn't take pcaps as input AFAIK for the CI fuzzing checks, just a corpus that it tests against. But, I think it may be possible to give it a pcap. Not sure how though. Philippe will be available next week to guide you better.
You can take a look at https://github.com/OISF/suricata/blob/master/src/tests/fuzz/README for building and running them locally